packages/devtools_app/lib/src/connected_app.dart

```diff
@@ -8,36 +8,66 @@ import 'globals.dart';
 import 'service_extensions.dart' as extensions;
 
 const flutterLibraryUri = 'package:flutter/src/widgets/binding.dart';
-const flutterWebLibraryUri = 'package:flutter_web/src/widgets/binding.dart';
 const dartHtmlLibraryUri = 'dart:html';
 
 class ConnectedApp {
   ConnectedApp();
 
-  Future<bool> get isFlutterApp async =>
-      _isFlutterApp ??= await _libraryUriAvailable(flutterLibraryUri);
+  bool get appTypeKnown => _appTypeKnown;
+  bool _appTypeKnown = false;
+
+  Future<bool> get isFlutterApp async {
+    _isFlutterApp ??= await _libraryUriAvailable(flutterLibraryUri);
+    _updateAppTypeKnown();
+    return _isFlutterApp;
+  }
+
+  bool get isFlutterAppRaw {
+    assert(_isFlutterApp != null);
+    return _isFlutterApp;
+  }
+
   bool _isFlutterApp;
 
-  Future<bool> get isPackageFlutterWeb async =>
-      _isPackageFlutterWeb ??= await _libraryUriAvailable(flutterWebLibraryUri);
-  bool _isPackageFlutterWeb;
+  Future<bool> get isProfileBuild async {
+    _isProfileBuild ??= await _connectedToProfileBuild();
+    _updateAppTypeKnown();
+    return _isProfileBuild;
+  }
+
+  bool get isProfileBuildRaw {
```

Comment:
`RemoteDiagnosticsNode` uses the convention of adding the suffix `Now` instead of  `Raw` to indicate getters that return a value immediately as opposed to matching getters that return a Future. I'm also open to using a different naming convention but if we do, we should be consistent across the application.

Reply:
changed to now